### ISSUE TITLE
Assume DoNotTrack by default

### ISSUE THREAD

**Issue by ldesgrange** [AmauriC/tarteaucitron.js#485]
Hi,

As a user I'm fed up with all the cookie consent splash screens opening when I visit most websites.

As a website administrator, from what I understand so far, if I configure TaC with `highPrivacy=true` and `handleBrowserDNTRequest=true`, a user coming to my website with `DNT=1` will not have TaC splash screen displayed, services not requiring consent (like matomo high privacy) are accepted and services requiring consent (like youtube) are denied.

I'm trying to find a way to have this behavior (no cookie consent splash screen, services not requiring consent are accepted, services requiring consent are refused) to be the default behavior for all the visitors of my website (as if DNT was turned on for all of them), but I haven't found a way to achieve that so far. Is there a way to configure TaC that way?

**Comment by alanmac** [comment]
I don't think DNT in its current form is an appropriate signal to achieve this. Perhaps for certain purposes (e.g. marketing / statistics), but what about additional website functionality that may require cookies that are inscope of requiring consent (e.g. functional cookies). Would it be appropriate not to give the user the opportunity to make an informed decision about these functionalities, rather than disable them all by default?

Banners can be annoying, but they do serve a purpose and the DNT functionality has been highlighted as potentially not appropriate, in that it basically bundles consent. I've [written about this here](https://www.trilateralresearch.com/cookie-guidance-data-protection-authorities-publish-updated/). Also see [DO NOT TRACK AND THE GDPR](https://www.w3.org/blog/2018/06/do-not-track-and-the-gdpr/).

Best,

Alan

**Comment by ldesgrange** [comment]
Thank you for this information (I haven't looked into the details but it seems that the British ICO and French CNIL do not have the same interpretation on which cookies require consent or not). Anyway my message was in fact not clear enough in the way that I don't think DNT is the appropriate tool either, but it's what at the moment allows me to achieve the behavior I'm looking for.

What I like in TaC, is that I can have the consent denied by default but then give it when I want to (for instance if I embed a Youtube video, the consent is asked the first time I want to see the video, and I think this behavior is much better on a user point of view).

Of course this behavior does not cover the case of cookies that require consent but are not linked to any user action, but it's also not the case I'm looking at.

**Comment by alanmac** [comment]
You're right, there can be slightly different guidance on which cookies require consent (e.g. the Spanish DPA currently suggests that Analytics does not require cosnent), but in general, there is broad agreement. What is not up for interpretation is the definition of consent - i.e. freely given, **specific**, informed, and an unambiguous indication of the individuals wishes (see [EDPB Guidelines on Consent](https://edpb.europa.eu/sites/edpb/files/files/file1/edpb_guidelines_202005_consent_en.pdf)). **Specific** is key here, and where DNT is concerned, means that it could not meet the standard of consent for multiple purposes. Consent needs to be unbundled.

As you suggest, where it concerns website features, a just-in-time consent model can work, asking the user to provide their consent at the point they require a feature. I think this is prefereable on many levels. It provides for a better user experience and allows the user to contextualise why they might want to consent to such a feature. In many cases though, this may not fully remove the need to have an initial banner for transparency purposes when the user first visits the site (e.g. in cases where some cookies are placed by default for which they can opt out of). But in some cases, when implemented correctly and where only 'strictly necessary' cookies are placed by default, a just-in-time consent model could indeed negate the need to display a banner on landing.

I believe you're right regarding a custom implementation of TaC by the CNIL. Note that the CNIL only have analytics cookies placed by default because they are likely using first party analytics, fully under their control. The same may not apply for implementations like Google Analytics.

**Comment by AmauriC** [comment]
Untested but it should work if you define window.doNotTrack == 1 before loading tarteaucitron

**Comment by ldesgrange** [comment]
> Untested but it should work if you define window.doNotTrack == 1 before loading tarteaucitron

Thanks for the tip, I did try that but it seems that the web browser don't let me change this value.
